lib/ex_aws/credentials_ini/file.ex

```diff
@@ -1,17 +1,111 @@
 if Code.ensure_loaded?(ConfigParser) do
   defmodule ExAws.CredentialsIni.File do
+    import ExAws.Request.Hackney, only: [request: 4]
     # as per https://docs.aws.amazon.com/cli/latest/topic/config-vars.html
     @valid_config_keys ~w(
       aws_access_key_id aws_secret_access_key aws_session_token region
       role_arn source_profile credential_source external_id mfa_serial role_session_name credential_process
+      sso_start_url sso_region sso_account_id sso_role_name
     )
 
     def security_credentials(profile_name) do
       shared_credentials = profile_from_shared_credentials(profile_name)
       config_credentials = profile_from_config(profile_name)
+      merge_credentials(config_credentials, shared_credentials)
+    end
+
+    defp merge_credentials(
+           %{
+             sso_start_url: sso_start_url,
+             sso_account_id: sso_account_id,
+             sso_role_name: sso_role_name,
+           } = config_credentials,
+           _shared_credentials
+         ) do
+           case get_sso_role_credentials(sso_start_url, sso_account_id, sso_role_name) do
+             {:ok, sso_creds} -> {:ok, Map.merge(sso_creds, config_credentials)}
+             {:error, _} = err -> err
+           end
+    end
+
+    defp merge_credentials(config_credentials, shared_credentials) do
       Map.merge(config_credentials, shared_credentials)
     end
 
+    defp get_sso_cache_file(sso_start_url) do
+      hash = :crypto.hash(:sha, sso_start_url) |> Base.encode16() |> String.downcase()
+
+      System.user_home()
+      |> Path.join(".aws/sso/cache/#{hash}.json")
+    end
+
+    defp get_sso_role_credentials(sso_start_url, sso_account_id, sso_role_name) do
+      sso_start_url
+      |> get_sso_cache_file()
+      |> File.read()
+      |> parse_sso_cache_file
+      |> check_sso_expiration
+      |> request_sso_role_credentials(sso_account_id, sso_role_name)
+      |> rename_sso_credential_keys
+    end
+
+    defp parse_sso_cache_file({:ok, contents}) do
+      contents
+      |> Jason.decode()
+    end
+
+    defp parse_sso_cache_file(err), do: err
+
+    defp check_sso_expiration({:ok, %{"expiresAt" => expires_at_str} = config}) do
+      with {:ok, expires_at, _} <- DateTime.from_iso8601(expires_at_str)
+      do
+        case DateTime.compare(expires_at, DateTime.utc_now()) do
+          :gt -> {:ok, config}
+          _ -> {:error, "SSO access token is expired, refresh the token with `aws sso login`"}
+        end
+      else
+        {:error, err} -> {:error, "SSO cache file has invalid expiration format: #{err}"}
+      end
+    end
```

Comment:
doesn't really need to return the config, could just return `:ok` or the error and reason